src/workerd/util/sqlite.c++

```diff
@@ -992,6 +998,10 @@ SqliteDatabase::Query::~Query() noexcept(false) {
       sqlite3_clear_bindings(&statement);
     }
   }
+  // Update Query stats only if db is not reset
```

Comment:
This comment implies that if you reset the database, the rows you've already read or written will not be accounted for.  I think you may need to add something to `SqliteDatabase::Query::beforeSqliteReset()` to record this information.

Reply:
Hmm, we do need it in there. I just have to make sure we don't run into race condition where the rows gets updated from the beforeSqliteReset and from the destructor